hotsos/core/host_helpers/config.py

```diff
@@ -100,62 +97,61 @@ def bool_str(val):
         return val
 
     @property
-    def all(self):
-        return self._sections
+    def all_keys(self):
+        return [x for option in self.config.items()
+                for x in list(option[1].keys())]
 
     def get(self, key, section=None, expand_to_list=False):
-        """ If section is None use flattened """
-        if section is None:
-            value = self._flattened_config.get(key)
-        else:
-            if section not in self._sections:
-                log.debug("section '%s' not found in config file, "
-                          "trying lower case", section)
-                section = section.lower()
-
-            value = self._sections.get(section, {}).get(key)
-
-        if expand_to_list:
-            return self.expand_value_ranges(value)
-
-        return value
-
-    @property
-    def dump(self):
-        with open(self.path) as fd:
-            return fd.read()
+        """ If section is None, then search all sections and return first
+        match. """
+        log.debug("ini read call -> %s:%s", section, key)
+
+        if not self.config:
+            return None
+
+        def post_processing(v):
```

Comment:
inline functions are hard to test, can we move this outside please?

Reply:
this inline function is just a named lambda in reality, but fair enough.

Reply:
done